components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/signup/UserSelfRegistrationManager.java

```diff
@@ -1041,8 +1041,8 @@ public void confirmVerificationCodeMe(String code, Map<String, String> propertie
                 }
             } else {
                 userClaims.put(IdentityRecoveryConstants.MOBILE_NUMBER_CLAIM, pendingMobileNumberClaimValue);
+                userClaims.put(NotificationChannels.SMS_CHANNEL.getVerifiedClaimUrl(), Boolean.TRUE.toString());
```

Comment:
Let's check in the MOBILE_VERIFICATION_ON_VERIFIED_LIST_UPDATE, if we can check the existing primary value and if the same get updated, we can set the verifiedMobile claim as `true`.

Reply:
Resolved via https://github.com/wso2-extensions/identity-governance/pull/948/commits/d5ef792fbc5ca807654f313f0e92ca1368ec9194